firedrake/projection.py

```diff
@@ -20,6 +24,7 @@
 def project(v, V, bcs=None, mesh=None,
```

Comment:
Since you remove the call to `_solve` can you remove it (and the comment above) just here?  @dham is testing the pyadjoint (i.e. new firedrake-adjoint) with py3, so plausibly we will jump ship, which will stop this branch failing tests.